src/Interval.hs

```diff
@@ -69,94 +69,94 @@ interval <-> x =  interval <+> (-x)
 
 
 raisePerfect :: IQuality -> IQuality
-raisePerfect IPerfect = (IAugmented 1)
-raisePerfect (IAugmented x) = (IAugmented (x+1))
+raisePerfect IPerfect = IAugmented 1
+raisePerfect (IAugmented x) = IAugmented $ x + 1
 raisePerfect (IDiminished 1) = IPerfect
-raisePerfect (IDiminished x) = (IDiminished (x-1)) 
+raisePerfect (IDiminished x) = IDiminished $ x - 1
 
 raiseMajor :: IQuality -> IQuality
-raiseMajor IMajor = (IAugmented 1)
-raiseMajor (IAugmented x) = (IAugmented (x+1))
+raiseMajor IMajor = IAugmented 1
+raiseMajor (IAugmented x) = IAugmented $ x + 1
 raiseMajor IMinor = IMajor
 raiseMajor (IDiminished 1) = IMinor
-raiseMajor (IDiminished x) = (IDiminished (x-1))
+raiseMajor (IDiminished x) = IDiminished $ x - 1
 
 lowerPerfect :: IQuality -> IQuality
-lowerPerfect IPerfect = (IDiminished 1)
-lowerPerfect (IDiminished x) = (IDiminished (x+1)) 
+lowerPerfect IPerfect = IDiminished 1
+lowerPerfect (IDiminished x) = IDiminished $ x + 1
 lowerPerfect (IAugmented 1) = IPerfect
-lowerPerfect (IAugmented x) = (IAugmented (x-1))
+lowerPerfect (IAugmented x) = IAugmented $ x-1
 
 lowerMajor :: IQuality -> IQuality
 lowerMajor IMajor = IMinor
-lowerMajor IMinor = (IDiminished 1)
-lowerMajor (IDiminished x) = (IDiminished (x+1))
+lowerMajor IMinor = IDiminished 1
+lowerMajor (IDiminished x) = IDiminished $ x + 1
 lowerMajor (IAugmented 1) = IMajor
-lowerMajor (IAugmented x) = (IAugmented (x-1))
+lowerMajor (IAugmented x) = IAugmented $ x - 1
 
 
 defaultIQuality :: Int -> IQuality
 defaultIQuality i =
-  case (intMod i) of
+  case intMod i of
     intervalInt
       | intervalInt `elem` [1, 4, 5] -> IPerfect
       | intervalInt `elem` [2, 3, 6, 7] -> IMajor
       | otherwise -> error "Impl error, mod 7 issue"
- 
+
 
 modShift :: Int -> Int -> (Int -> Int)
-modShift newZero modulus = (\i -> ((i - newZero) `mod` modulus) + newZero)
+modShift newZero modulus i = ((i - newZero) `mod` modulus) + newZero
```

Comment:
You had some clear semantics in mind here, but it's good to get into the habit of thinking of curried functions as, well, curried functions :) Anytime you have a lambda to the right of a definition, that lambda can be eliminated by making its parameter a parameter to the name you're binding (in this case, `modShift`.)